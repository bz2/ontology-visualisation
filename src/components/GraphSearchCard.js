// import { useState } from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { Button } from 'primereact/button'
import { Card } from 'primereact/card'
import actions from '../store/actions'
import setSearchNeighbourNodes from '../utils/graphSearch/setSearchNeighbourNodes'
import { OPERATION_TYPE_UPDATE } from '../constants/store'
import { SIDEBAR_VIEW_NOTES, SIDEBAR_VIEW_SYNONYMS } from '../constants/views'
import { DATASET_REPO_URL } from '../constants/routes'

const GraphSearchCard = ({
  updateStoreValue,
  searchResult,
}) => {
  const { t } = useTranslation()

  const {
    type,
    name,
    id,
    skosDefinition,
    skosComment,
    description,
    path
  } = searchResult

  const infoDescription = type === 'dataset' ? description : (
    skosDefinition || skosComment
  )

  return (
    <Card>
      <div className="p-card-info">
        <div className="p-card-row bold">
          {type === 'dataset' ? t(type) : t('dataEntity')}
        </div>

        <div className="p-card-row p-card-info-title">
          {name}
        </div>

        <p className="p-text-justify">
          {infoDescription}
        </p>
      </div>

      <div className="p-card-buttons">
        <Button
          aria-label={t('visualise')}
          label={t('visualise')}
          onClick={() => setSearchNeighbourNodes({
            separationDegree: 1,
            updateStoreValue,
            searchResult,
          })}
        />

        {type === 'dataset' ? (
          <a
            href={`${DATASET_REPO_URL}${path}`}
            aria-label={t('open')}
            className="p-button p-text-center link"
            rel="noopener nofollow noreferrer"
            target="_blank"
          >
            {t('open')}
          </a>
        ) : (
          <>
            <Button
              aria-label={t('notes')}
              label={t('notes')}
              onClick={() => {
                setSearchNeighbourNodes({
                  separationDegree: 1,
                  updateStoreValue,
                  searchResult,
                })

                updateStoreValue(['selectedElement'], OPERATION_TYPE_UPDATE, { [id]: 'node' })
                updateStoreValue(['sidebarView'], OPERATION_TYPE_UPDATE, SIDEBAR_VIEW_NOTES)
              }}
            />

            <Button
              aria-label={t('synonyms')}
              label={t('synonyms')}
              onClick={() => {
                setSearchNeighbourNodes({
                  separationDegree: 1,
                  updateStoreValue,
                  searchResult,
                })

                updateStoreValue(['selectedElement'], OPERATION_TYPE_UPDATE, { [id]: 'node' })
                updateStoreValue(['sidebarView'], OPERATION_TYPE_UPDATE, SIDEBAR_VIEW_SYNONYMS)
              }}
            />
          </>
        ) }

      </div>
    </Card>
  )
}

GraphSearchCard.propTypes = {
  updateStoreValue: PropTypes.func.isRequired,
  searchResult: PropTypes.shape().isRequired,
}

const mapToProps = ({
  entrySearchResults,
}) => ({
  entrySearchResults,
})

export default connect(
  mapToProps,
  actions
)(GraphSearchCard)
