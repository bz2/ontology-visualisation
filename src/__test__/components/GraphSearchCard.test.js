import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import GraphSearchCard from '../../components/GraphSearchCard'

const setup = () => {
  const props = {
    updateStoreValue: jest.fn(),
    searchResult: {
      id: '123',
      rdfsLabel: 'test',
      rdfAbout: 'http://test.test/test'
    },
    stylingNodeCaptionProperty: 'rdfsLabel',
    stylingEdgeCaptionProperty: 'rdfsLabel'
  }

  const component = shallow(<GraphSearchCard {...props} />)

  return {
    component,
    props
  }
}

describe('GraphSearchCard', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
