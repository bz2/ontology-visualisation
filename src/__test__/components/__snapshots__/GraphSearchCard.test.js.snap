// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphSearchCard should match snapshot 1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
      "toggleFromArrayInKey": [Function],
      "toggleFromSubArray": [Function],
    }
  }
  addToObject={[MockFunction]}
  mapToProps={[Function]}
  searchResult={
    Object {
      "id": "123",
      "rdfAbout": "http://test.test/test",
      "rdfsLabel": "test",
    }
  }
  setStoreState={[MockFunction]}
  stylingEdgeCaptionProperty="rdfsLabel"
  stylingNodeCaptionProperty="rdfsLabel"
>
  <Component />
</Connect>
`;
